lib/charms/mongodb/v1/mongodb_backups.py

```diff
@@ -806,3 +819,35 @@ def get_backup_error_status(self, backup_id: str) -> str:
                 return backup.get("error", "")
 
         return ""
+
+    def are_s3_configurations_provided(self) -> bool:
+        """Returns True if not all necessary configurations for s3 are provided.
+
+        Minimum necessary s3 credentials can be found by reading the PBM page:
+        https://docs.percona.com/percona-backup-mongodb/reference/configuration-options.html#s3-type-storage-options
+        """
+        if not self.model.get_relation(S3_RELATION):
+            logger.info("No configurations for backups, not relation to s3-charm.")
+            return False
+
+        provided_configs = self._get_pbm_configs()
+        if (
+            "storage.s3.credentials.access-key-id" not in provided_configs
+            or "storage.s3.credentials.secret-access-key" not in provided_configs
+        ):
+            logger.info("Missing s3 credentials")
+            return False
+
+        if "storage.s3.bucket" not in provided_configs:
+            logger.info("Missing bucket")
+            return False
+
+        # since we cannot determine whether the user has an AWS or GCP bucket or Minio bucket
+        # send them an info
+        if "storage.s3.region" not in provided_configs:
+            logger.info("Missing region - this is required for AWS and GCP")
+
+        if "storage.s3.endpointUrl" not in provided_configs:
+            logger.info("Missing endpointUrl - this is required for MinIO and GCP")
```

Comment:
Can this info not be inferred / deducted from the previous config options?

Reply:
now that I think of it, I think it might be possible to determine which provider we use by parsing the format of the access key and see if its format is that of AWS or GCP. (but that assumes that the user has provided correct credentials - and would be a little hacky.) Lmk what you think here, I trust your vision.

Reply:
Let's address it in another PR